code/modules/admin/tag.dm

```diff
@@ -0,0 +1,128 @@
+/**
+ * Creates a weakref of the target datum and inserts it into [/datum/admins/var/tagged_datums], for later reference.
+ *
+ * Note that if the original target is deleted and another datum is put in the same memory address, tthis
+ *
+ * Arguments:
+ * * target_datum - The datum you want to create a tag for. Should be the datum itself, not a weakref
+ */
+/datum/admins/proc/add_tagged_datum(datum/target_datum)
+	var/datum/weakref/new_weakref = WEAKREF(target_datum)
+
+	if(LAZYFIND(tagged_datums, new_weakref))
+		new_weakref = null // needed?
+		to_chat(owner, span_warning("[target_datum] is already tagged!"))
+		return
+
+	LAZYADD(tagged_datums, new_weakref)
+	RegisterSignal(target_datum, COMSIG_PARENT_QDELETING, .proc/handle_tagged_del)
```

Comment:
(If you do change to just refs do remember to clear the tagged_datums list on admin holder qdel, since the signals will just cease to exist but the list may continue on